### ISSUE TITLE
feat(metallicity): serve the published classifier as a prediction

### ISSUE THREAD

**Issue by junwen94** [stfc/goldilocks-ml#15]
## Problem

There is a metallicity classifier in this package, and this package cannot
predict metallicity.

`models/metallicity/cgcnn/representation.py` ports the published CGCNN
(`ptc95-vbq12`) faithfully, classification head included. `CGCNN.forward`
returns metallicity class logits. Nothing in the repository calls it:

```
class CGCNN
  forward()               -> metallicity class logits     never called
  extract_crystal_repr()  -> the 64-wide pooled vector    the only caller

crystal_representations()  the module's only public entry point
__init__.py exports        crystal_representations
```

The model was ported for one purpose — supplying 64 columns to the QRF95
feature vector — and its actual output was left unwired. The module name says
so. Using an intermediate layer as a transfer feature is a legitimate
technique, but it is a secondary use of this artifact, and it is currently the
only one.

The consequence reaches Core. `goldilocks_core/analysis.py` decides electronic
character with a rule:

```python
if elements and all(element.is_metal for element in elements):
    -> "likely_metal"
```

Every element being a metal is not the same claim as the compound being one.
`ElectronicCharacter` already admits `metal`, `insulator`, `likely_metal`, and
`unknown`, so the vocabulary for a real answer exists and nothing fills it —
while a trained classifier for exactly this question sits published on PSDI and
vendored in this repository.

### Serve the classifier through the existing seam

Metallicity is a DFT-relevant property a model predicts from a structure, which
is what `goldilocks_ml.inference` is for. It needs the same three pieces the
k-distance model has:

- a target contract in `CONTRACTS`, naming the parameter and quantity — the
  quantity is a class label with a probability, not a scalar to convert, which
  is the first prediction of that shape and will exercise `ModelPrediction`'s
  `value` beyond `float`;
- a serving runtime `metallicity.cgcnn` registered as a predictor, loading the
  checkpoint and calling `forward`;
- a model record for the published checkpoint so `load_model` can read it, or
  a documented path for constructing one from the deposit.

The prediction should carry the class and its probability, with the probability
in `details` rather than in `value`, matching the decision that a consumer acts
on one value and records the rest.

### Correct the module's priorities

`representation.py` is named after its secondary output. The classifier is what
the module is. Rename it, export the prediction path from `__init__.py`, and
keep `crystal_representations` as an explicitly documented escape hatch for the
QRF95 feature contract, marked as coupling to network internals that the
published artifact does not promise to hold stable.

### Do not retrain anything

The published checkpoint is the model. Whether a better one is wanted is a
separate question, and #3 is where it belongs.

## Why this is worth doing before the work that depends on it

- **Core gets a real answer now.** Replacing the all-elements-are-metals rule
  needs no training, no new dataset, and no new deposit — only the serving path
  this issue adds.
- **#14 needs it.** The ablation there compares the 64-column block against the
  classifier's scalar probability, which cannot be computed today.
- **#3 stops being a prerequisite.** With serving in place, retraining becomes a
  question of whether the published classifier is good enough, not something
  Core waits on.

## Acceptance criteria

- [ ] `load_model` returns a metallicity model from the published checkpoint
- [ ] `predict` returns a `ModelPrediction` naming the parameter, with the class
      as its value and the probability in `details`
- [ ] A target contract for metallicity is registered, and an artifact
      declaring an unknown one is refused by name as k-distance artifacts are
- [ ] The module is named for the classifier; `crystal_representations` remains
      available and documented as a secondary use
- [ ] QRF95 feature extraction is unchanged and its tests still pass
- [ ] `docs/inference.md` covers a second parameter, showing the seam carries
      more than k-points

---
Written by an agent on behalf of Junwen Yin.
